### DISCUSSION TITLE
Using useAsyncData or useFetch

### DISCUSSION THREAD

**Discussion by cyriaque-ovaga** [nuxt/nuxt#18349] [Questions] [answered]
Hello I would like if I can use useAsyncData or useFetch in
<script>
export default {
};
</script>

Or 
Only in 
<script setup>
</script>

**Comment by lucpotage** [comment, marked as answer]
Composition API must be used in `setup()` as explained in Vue docs: https://v3.vuejs.org/guide/composition-api-introduction.html#basics-of-composition-api.

```vue
<script>
export default {
  setup() {
    // here
  }
}
</script> 
```

```vue
<script setup>
// or here
</script>
```

**Reply by cyriaque-ovaga** [reply]
Thank you. You make my day
